rest/route/project.go

```diff
@@ -656,6 +656,82 @@ func (h *projectRepotrackerHandler) Run(ctx context.Context) gimlet.Responder {
 	return gimlet.NewJSONResponse(struct{}{})
 }
 
+////////////////////////////////////////////////////////////////////////
+//
+// DELETE /rest/v2/projects/{project_id}
+
+type projectDeleteHandler struct {
+	projectName string
+	sc          data.Connector
+}
+
+func makeDeleteProject(sc data.Connector) gimlet.RouteHandler {
+	return &projectDeleteHandler{
+		sc: sc,
+	}
+}
+
+func (h *projectDeleteHandler) Factory() gimlet.RouteHandler {
+	return &projectDeleteHandler{
+		sc: h.sc,
+	}
+}
+
+func (h *projectDeleteHandler) Parse(ctx context.Context, r *http.Request) error {
+	h.projectName = gimlet.GetVars(r)["project_id"]
+	return nil
+}
+
+func (h *projectDeleteHandler) Run(ctx context.Context) gimlet.Responder {
+	project, err := dbModel.FindMergedProjectRef(h.projectName)
+	if err != nil {
+		return gimlet.MakeJSONErrorResponder(err)
+	}
+
+	if project.Hidden {
+		return gimlet.MakeJSONErrorResponder(errors.Errorf("project '%s' is already hidden", h.projectName))
+	}
+
+	if !project.UseRepoSettings {
+		return gimlet.MakeJSONErrorResponder(
+			errors.Errorf("project '%s' must have UseRepoSettings enabled to be eligible for deletion", h.projectName))
+	}
+
+	skeletonProj := dbModel.ProjectRef{
+		Id:              project.Id,
+		Owner:           project.Owner,
+		Repo:            project.Repo,
+		Branch:          project.Branch,
+		RepoRefId:       project.RepoRefId,
+		Enabled:         false,
+		UseRepoSettings: true,
+		Hidden:          true,
+	}
+	if updateErr := skeletonProj.Update(); updateErr != nil {
+		return gimlet.MakeJSONErrorResponder(updateErr)
+	}
+
+	projectAliases, err := dbModel.FindAliasesForProject(h.projectName)
```

Comment:
You'll want to use project.Id -- technically users can pass in Id or Identifier for project routes, but the alias/variable IDs correspond to project.Id (because Id immutable, while Identifier can be modified)